api/notd/token_listing_processor.py

```diff
@@ -28,10 +29,57 @@ def _parse_date_string(dateString: str) -> datetime.datetime:
 
 class TokenListingProcessor:
 
-    def __init__(self, requester: Requester, openseaRequester: Requester, lockManager: LockManager):
+    def __init__(self, requester: Requester, openseaRequester: Requester, lockManager: LockManager, collectionManger: CollectionManager):
         self.requester = requester
         self.openseaRequester = openseaRequester
         self.lockManager = lockManager
+        self.collectionManger = collectionManger
+
+    async def get_opensea_listings_for_collection(self, registryAddress: str) -> List[RetrievedTokenListing]:
+        listings = []
+        async with self.lockManager.with_lock(name='opensea-requester', timeoutSeconds=100, expirySeconds=240):
+            collection = await self.collectionManger.get_collection_by_address(address=registryAddress)
+            collectionOpenseaSlug = collection.openseaSlug
+            nextPageId: Optional[str] = None
+            pageCount = 0
+            while True:
+                logging.stat('RETRIEVE_LISTINGS_OPENSEA', registryAddress, float(f'{pageCount}'))
+                queryData: Dict[str, JSON1] = {
+                    "limit": 100
+                }
+                if nextPageId:
+                    queryData['next'] = nextPageId
+                response = await self.openseaRequester.get(url=f'https://api.opensea.io/v2/listings/collection/{collectionOpenseaSlug}/all', dataDict=queryData,timeout=30)
+                responseJson = response.json()
+                for openseaListing in (responseJson.get('listings') or []):
+                    startDate = datetime.datetime.utcfromtimestamp(int(openseaListing['protocol_data']["parameters"]["startTime"]))
```

Comment:
maybe we can extract this code into a private function cos its copy pasted in the other OS function right?